test/staged.jl

```diff
@@ -156,7 +156,7 @@ end
 @generated function _g_f_with_inner(x)
     :(y->y)
 end
-@test (_g_f_with_inner(1))(8) == 8
+@test_throws ErrorException _g_f_with_inner(1)
 
```

Comment:
At some point, we likely may want to reintroduce this, but right now it gets handled incorrectly by the runtime system and will likely require not-backportable changes to fix.

> What's allowed in generated functions is starting to become a pretty restrictive subset of the language

It's not "starting to become" because it's really not a new restriction, it's just better at enforcing correctness up-front now. On the other hand, the fix for #265 *will* add significant new restrictions to them, although it shouldn't affect anyone who's been fully observant of their definition.

> and I don't think people have been aware of that or writing them that way.

That's too bad, but the manual does repeat several times that "you should /never/ write a generated function with side effects" [emphasis original] http://docs.julialang.org/en/latest/manual/metaprogramming/#generated-functions (unfortunately interspersed with various buggy examples). I can repeat it a few more times, but I consider it better just to enforce the restrictions.